src/code/z_fbdemo_fade.c

```diff
@@ -111,13 +115,13 @@ void TransitionFade_SetColor(void* thisx, u32 color) {
 void TransitionFade_SetType(void* thisx, s32 type) {
     TransitionFade* this = (TransitionFade*)thisx;
 
-    if (type == 1) {
+    if (type == TRANS_INSTANCE_TYPE_FILL_OUT) {
         this->fadeType = 1;
         this->fadeDirection = 1;
-    } else if (type == 2) {
+    } else if (type == TRANS_INSTANCE_TYPE_FILL_IN) {
         this->fadeType = 1;
         this->fadeDirection = 0;
```

Comment:
Could you explain the difference between `type`, `fadeType` and `fadeDirection`, and also why `z_fbdemo_triforce.c` has the `TRANS_INSTANCE_TYPE_` enum on the switch for `this->fadeDirection`?

Reply:
`type` is specifically what `type` of transition is being set, much like how an actor's params can tell what `type` of the actor should be used. 

From https://github.com/zeldaret/oot/blob/b23270180b49245e6f7bf82632af52b06360e6c6/src/code/z_play.c#L611-L615 `types` 1 and 2 is generic and applies to each and every transition. Generally this type just means what direction the transition should go (from play to the transition, or from the transition to play) and sets `fadeDirection`. `Triforce` actually uses the `type` value directly (https://github.com/zeldaret/oot/blob/b23270180b49245e6f7bf82632af52b06360e6c6/src/code/z_fbdemo_triforce.c#L56-L60) which is why I used the enum (though on second thought should probably change the member name to `type` or something). `Wipe1` and `Circle` does a small conversion (just converts 2 to 0) for `direction` (perhaps just a boolean like `directionIn` for this 🤔 instead of an enum). `Fade` is unique in that it doesn't have just the two types, but adds one for special occasions (flash). So `fade` has an additional `fadeType` (which is purely internal) to distinguish between this additional type and your normal transition fades, along with the usual `fadeDirection` that all other transitions have for the regular transitions to tell which direction the transition should go.

Reply:
I'm curious, what are people's thoughts on changing `fadeDirection/direction` used in most transitions to a boolean?

Reply:
Thanks for this explanation! It makes more sense now